geonode_mapstore_client/templatetags/get_menu_json.py

```diff
@@ -12,96 +16,100 @@ def _handle_single_item(menu_item):
         m_item['target'] = '_blank'
     return m_item
 
-@register.simple_tag
-def get_base_left_topbar_menu():
 
-    return [
-        {
-            "label": "Datasets",
-            "type": "dropdown",
-            "items": [
-                {
-                    "type": "link",
-                    "href": "/catalogue/#/search/?filter{resource_type.in}=layer",
-                    "label": "Datasets"
-                },
-                {
-                    "type": "divider"
-                },
+@register.simple_tag(takes_context=True)
+def get_base_left_topbar_menu(context):
+    datasets = {
+        "label": "Datasets",
+        "type": "dropdown",
+        "items": [
+            {
+                "type": "link",
+                "href": "/catalogue/#/search/?filter{resource_type.in}=layer",
+                "label": "Datasets"
+            }
+        ]
+    }
+
+    documents = {
+        "label": "Documents",
+        "type": "dropdown",
+        "items": [
+            {
+                "type": "link",
+                "href": "/catalogue/#/search/?filter{resource_type.in}=document",
+                "label": "Documents"
+            }
+        ]
+    }
+    devider = {
+        "type": "divider"
+    }
+    maps = {
+        "label": "Maps",
+        "type": "dropdown",
+        "items": [
+            {
+                "type": "link",
+                "href": "/catalogue/#/search/?filter{resource_type.in}=map",
+                "label": "Explore maps"
+            }
+        ]
+    }
+    geostories = {
+        "label": "GeoStories",
+        "type": "dropdown",
+        "items": [
+            {
+                "type": "link",
+                "href": "/catalogue/#/search/?filter{resource_type.in}=geostory",
+                "label": "GeoStories"
+            }
+        ]
+    }
+    user = context.get('request').user
+    if user.is_authenticated:
```

Comment:
just a note all the following link should be enabled only if user has add resource permissions:

- Upload dataset
- Upload document
- Create map
- Create GeoStory

Reply:
Missing also badge property with the total count of resources by type